build-tasks/templates/rule.template.js

```diff
@@ -1,42 +1,37 @@
-module.exports = ({ ruleName, walkerName }) =>
+module.exports = ({ ruleName, walkerName, type, description, issueClass, issueType, severity, level, group }) =>
     `import * as ts from 'typescript';
 import * as Lint from 'tslint';
 
-import {ExtendedMetadata} from './utils/ExtendedMetadata';
-// use (and contribute to) AstUtils for common AST functions // TODO: delete comment
-import {AstUtils} from './utils/AstUtils';
-// use Utils instead of Underscore functions // TODO: delete comment
-import {Utils} from './utils/Utils';
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { AstUtils } from './utils/AstUtils';
+import { Utils } from './utils/Utils';
 
 const FAILURE_STRING: string = 'Some error message: '; // TODO: Define an error message
 
 export class Rule extends Lint.Rules.AbstractRule {
-
     public static metadata: ExtendedMetadata = {
         ruleName: '${ruleName}',
-        type: 'maintainability',    // one of: 'functionality' | 'maintainability' | 'style' | 'typescript'
-        description: '... add a meaningful one line description',
+        type: '${type}',
+        description: '${description}',
         options: null, // tslint:disable-line:no-null-keyword
         optionsDescription: '',
-        optionExamples: [],         // Remove this property if the rule has no options
+        optionExamples: [], // TODO: Remove this property if the rule has no options
         typescriptOnly: false,
```

Comment:
Would be nice to ask this during setup.
There is issue #646 to convert old rules. And I think that new rules should support JS from the begging if applicable.